source/com.microsoft.tfs.client.common/src/com/microsoft/tfs/client/common/credentials/CredentialsHelper.java

```diff
@@ -161,4 +183,103 @@ public static TFSConnection getVstsRootConnection(final TFSConnection connection
 
         return vstsConnection;
     }
+
+    public static PatCredentials getOAuthCredentials(final URI serverURI) {
+        final Authenticator authenticator;
+        final Options options = Options.getDefaultOptions();
+        options.patGenerationOptions.displayName = PatCredentials.USERNAME_FOR_CODE_ACCESS_PAT;
+        options.patGenerationOptions.tokenScope = VsoTokenScope.AllScopes;
+
+        if (serverURI != null) {
+            log.debug("Interactively retrieving credential based on oauth2 flow for " + serverURI.toString()); //$NON-NLS-1$
+            log.debug("Trying to persist credential, generating a PAT"); //$NON-NLS-1$
+
+            /*
+             * If this credential is to be persisted, then let's create a PAT
+             */
+            authenticator = new VstsPatAuthenticator(CLIENT_ID, REDIRECT_URL, accessTokenStore, tokenStore);
+        } else {
+            log.debug("Interactively retrieving credential based on oauth2 flow for VSTS"); //$NON-NLS-1$
+            log.debug("Do not try to persist, generating oauth2 token."); //$NON-NLS-1$
+
+            /*
+             * Not persisting this credential, simply create an oauth2 token
+             */
+            authenticator = OAuth2Authenticator.getAuthenticator(CLIENT_ID, REDIRECT_URL, accessTokenStore);
+        }
+
+        final UserPasswordCredentialProvider provider = new UserPasswordCredentialProvider(authenticator);
+
+        Credential tokenCreds;
+
+        tokenCreds = provider.getCredentialFor(
+            serverURI != null ? serverURI : URIUtils.VSTS_ROOT_URL,
+            PromptBehavior.AUTO,
+            options);
+
+        if (tokenCreds == null) {
+            // OAuth2 token is probably expired. Remove it from the internal
+            // memory store...
+            accessTokenStore.delete("OAuth2:" + URIUtils.VSTS_ROOT_URL_STRING); //$NON-NLS-1$
```

Comment:
It's probably better to call authenticator.signOut() which shields you from internal implementation details.